src/components/SessionActivityTracker.js

```diff
@@ -41,23 +36,32 @@ class SessionActivityTracker extends React.Component {
       this.setState({ counter: this.props.config.get('userSessionTimeoutInterval') })
     }
   }
+
   decrementCounter () {
-    if (this.state.counter !== null) {
-      const state = {
-        counter: this.state.counter - 1,
-      }
-      if (this.state.counter <= TIME_TO_DISPLAY_MODAL && !this.state.showTimeoutModal) {
-        state.showTimeoutModal = true
-      }
-      if (this.state.counter <= 0) {
+    const { counter } = this.state
+    if (counter !== null) {
+      this.setState({
+        counter: counter - 1,
+        showTimeoutModal: counter <= TIME_TO_DISPLAY_MODAL,
+      })
+
+      if (counter <= 0) {
         this.props.onLogout()
       }
-      this.setState(state)
     }
   }
 
+  componentDidMount () {
+    document.body.addEventListener('mousemove', this.dropCounter)
+    this.timer = setInterval(this.decrementCounter, 1000)
+  }
+
   componentWillUnmount () {
-    clearInterval(this.timer)
+    document.body.removeEventListener('mousemove', this.dropCounter)
+    if (this.timer) {
+      clearInterval(this.timer)
+      this.timer = null
```

Comment:
`componentWillUnmount` is last stage of component's life cycle, so after this call component will be destroyed, so this line is unnecessary.